### ISSUE TITLE
Scenedetect fails on mp4 video with multiple audio tracks

### ISSUE THREAD

**Issue by tslater** [Breakthrough/PySceneDetect#179]
**Bug/Issue Description:**
I get fatal errors using scenedetect with a multiaudio stream mp4. When I used ffmpeg to copy to an mkv with the same streams, or when I copy to an mp4 with only one audio stream, it works. It will not work with multiple streams in an mp4.

**Required Information:**
Provide the following information to assist with reporting the bug:
1. Provide a full copy of the command line options you are using, for example:

```
scenedetect --input /Volumes/sources/sources/video/movie/chapter_5/landscape.mp4  --output /Volumes/sources/sources/movie/movie/chapter_5/scene_detect  --stats stats.csv detect-content list-scenes save-images            

```

**attach the generated `BUG_REPORT.txt` file**.

```
INFO: __init__.scenedetect_cli(): Version: v0.5.3
INFO: __init__.scenedetect_cli(): Info Level: debug
INFO: __init__.scenedetect_cli(): Output directory set:
  /Volumes/sources/sources/video/movie/chapter_5/scene_detect
DEBUG: context.parse_options(): Parsing program options.
DEBUG: context._init_video_manager(): Initializing VideoManager.
INFO: video_manager.__init__(): Loaded 1 video, framerate: 23.98 FPS, resolution: 886 x 480
INFO: video_manager.set_downscale_factor(): Downscale factor set to 3, effective resolution: 295 x 160
DEBUG: context.parse_options(): VideoManager initialized.
DEBUG: __init__.detect_content_command(): Detecting content, parameters:
  threshold: 30, min-scene-len: 14
INFO: context.list_scenes_command(): Scene list CSV file name format:
  $VIDEO_NAME-Scenes.csv
INFO: context.save_images_command(): Image output format set: JPEG
DEBUG: context.process_input(): Processing input...
INFO: context.process_input(): Detecting scenes...
INFO: context.process_input(): Processed 0 frames in 0.0 seconds (average 0.00 FPS).
DEBUG: context.process_input(): No frame metrics updated, skipping update of the stats file.
INFO: context.process_input(): Detected 1 scenes, average shot length 0.0 seconds.
INFO: context.process_input(): Writing scene list to CSV file:
  /Volumes/sources/sources/video/movie/chapter_5/scene_detect/landscape-Scenes.csv
INFO: context.process_input(): Scene List:
-----------------------------------------------------------------------
 | Scene # | Start Frame |  Start Time  |  End Frame  |   End Time   |
-----------------------------------------------------------------------
 |      1  |           0 | 00:00:00.000 |           0 | 00:00:00.000 |
-----------------------------------------------------------------------

INFO: video_manager.set_downscale_factor(): Downscale factor set to 1, effective resolution: 886 x 480
INFO: context._generate_images(): Generating output images (3 per scene)...
DEBUG: context.cleanup(): Cleaning up...


INFO: __init__.scenedetect_cli(): Version: v0.5.3
INFO: __init__.scenedetect_cli(): Info Level: debug
INFO: __init__.scenedetect_cli(): Output directory set:
  /Volumes/sources/sources/video/movie/chapter_5/scene_detect
DEBUG: context.parse_options(): Parsing program options.
DEBUG: context._init_video_manager(): Initializing VideoManager.
INFO: video_manager.__init__(): Loaded 1 video, framerate: 23.98 FPS, resolution: 886 x 480
INFO: video_manager.set_downscale_factor(): Downscale factor set to 3, effective resolution: 295 x 160
DEBUG: context.parse_options(): VideoManager initialized.
DEBUG: __init__.detect_content_command(): Detecting content, parameters:
  threshold: 30, min-scene-len: 14
INFO: context.list_scenes_command(): Scene list CSV file name format:
  $VIDEO_NAME-Scenes.csv
INFO: context.save_images_command(): Image output format set: JPEG
DEBUG: context.process_input(): Processing input...
INFO: context.process_input(): Detecting scenes...
INFO: context.process_input(): Processed 0 frames in 0.0 seconds (average 0.00 FPS).
DEBUG: context.process_input(): No frame metrics updated, skipping update of the stats file.
INFO: context.process_input(): Detected 1 scenes, average shot length 0.0 seconds.
INFO: context.process_input(): Writing scene list to CSV file:
  /Volumes/sources/sources/video/movie/chapter_5/scene_detect/landscape-Scenes.csv
INFO: context.process_input(): Scene List:
-----------------------------------------------------------------------
 | Scene # | Start Frame |  Start Time  |  End Frame  |   End Time   |
-----------------------------------------------------------------------
 |      1  |           0 | 00:00:00.000 |           0 | 00:00:00.000 |
-----------------------------------------------------------------------

INFO: video_manager.set_downscale_factor(): Downscale factor set to 1, effective resolution: 886 x 480
INFO: context._generate_images(): Generating output images (3 per scene)...
DEBUG: context.cleanup(): Cleaning up...


```

**Expected Behavior:**
Expect the process to run with output and no error

**Computing Environment:**
Dockerfile
```
FROM python:3.8-slim

RUN apt-get update && apt-get install -y \
  libglib2.0-0 \
  libsm6 \
  libxext6 \
  libxrender-dev \
  libgl1-mesa-glx \
  && rm -rf /var/lib/apt/lists/*

RUN pip install --no-cache-dir scenedetect[opencv,progress_bar]==0.5.3
```

**Additional Information:**
Just my gratefulness for making this!

**Comment by tslater** [comment]
Here's the error omitted when I run without `BUG_REPORT.txt`:

```

[PySceneDetect] Downscale factor set to 1, effective resolution: 886 x 480
[PySceneDetect] Generating output images (3 per scene)...

  0%|          | 0/3 [00:00<?, ?images/s]Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 792, in main
    ctx.exit()
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 558, in exit
    raise Exit(code)
click.exceptions.Exit: 0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/scenedetect", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/scenedetect/__main__.py", line 58, in main
    cli.main(obj=cli_ctx)   # Parse CLI arguments with registered callbacks.
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 792, in main
    ctx.exit()
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 404, in __exit__
    self.close()
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 494, in close
    cb()
  File "/usr/local/lib/python3.8/site-packages/scenedetect/cli/context.py", line 395, in process_input
    self._generate_images(scene_list=scene_list, video_name=video_name,
  File "/usr/local/lib/python3.8/site-packages/scenedetect/cli/context.py", line 200, in _generate_images
    timecode_list = [
  File "/usr/local/lib/python3.8/site-packages/scenedetect/cli/context.py", line 201, in <listcomp>
    [
  File "/usr/local/lib/python3.8/site-packages/scenedetect/cli/context.py", line 202, in <listcomp>
    FrameTimecode(int(f), fps=fps) for f in [
  File "/usr/local/lib/python3.8/site-packages/scenedetect/frame_timecode.py", line 115, in __init__
    self.frame_num = self._parse_timecode_number(timecode)
  File "/usr/local/lib/python3.8/site-packages/scenedetect/frame_timecode.py", line 251, in _parse_timecode_number
    raise ValueError('Timecode frame number must be positive and greater than zero.')
ValueError: Timecode frame number must be positive and greater than zero.

  0%|          | 0/3 [00:00<?, ?images/s]
    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:314:20)
    at ChildProcess.EventEmitter.emit (domain.js:486:12)
    at maybeClose (internal/child_process.js:1051:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
```

**Comment by Breakthrough** [comment]
Hi @tslater;

Are you running into this issue with all files with multiple audio tracks, or just a few?  I'll try to replicate this on my end and keep you posted (may need a small sample if I'm unable to reproduce on my end).  Looks like you might've found a couple bugs here, thanks for the report.

One of my concerns is I'm not sure at first glance how I would go about detecting this condition at runtime, without relying on another tool like `ffprobe`, but will definitely do some more investigation on my end and see if I can come up with a resolution.  I haven't heard any issues like this before with OpenCV + video files, so hoping this might just be a one-off issue.

Thanks for the report!

**Comment by tslater** [comment]
You're very welcome. I'm seeing that it mkv videos with multiple audio tracks are working, but not mp4.  Hopefully that helps.

**Comment by tslater** [comment]
I can send an offending video file, but privately because the video files I'm working with are all copyrighted. I have permission to work with the film, but distribution rights are currently in negotiation status.

**Comment by Breakthrough** [comment]
@tslater If you could that would be great.  Send me a message via http://bcastell.com/contact/ and I'll get in touch with you.  Thanks!

**Comment by Breakthrough** [comment]
Hey @tslater;

Should have some improved messaging for this error now in v0.5.4, so this should be in the next release.  Long term would require switching libraries to handle video I/O which isn't really feasible right now unfortunately.  I did update the FAQ with two methods to get around this issue using `ffmpeg` or `mkvmerge` via the following commands:

    ffmpeg -i input.mp4 -c copy -an output.mp4

    mkvmerge -o output.mkv input.mp4

Thanks again for the report!

**Comment by Breakthrough** [comment]
This may actually be possible to fix once multiple video backends are supported as per #213 - that is still a way's away, but just wanted to state that for completeness.

**Comment by Breakthrough** [comment]
Interestingly enough, I'm now unable to reproduce this issue with the latest version of Python/OpenCV.  If anyone runs into this issue, try updating to the latest version and see if that resolves the issue - and if not, please post an update here!

**Comment by Breakthrough** [comment]
Some users are still running into issues with this - this should be resolved in v0.6 if you install PyAV (#213).  An error message will also be displayed when processing a video fails indicating to attempt using the PyAV backend as an alternative.
